src/search.cpp

```diff
@@ -346,14 +343,10 @@ void Thread::search() {
   bestValue = delta = alpha = -VALUE_INFINITE;
   beta = VALUE_INFINITE;
 
-  if (mainThread)
+  if (mainThread && mainThread->previousScore != VALUE_INFINITE)
   {
```

Comment:
patch looks good to me, the curly brackets can be removed here.

Reply:
Hmm, after an if statement, we usually have the curly brackets in place, no?

Reply:
usually not if it is followed by a single statement (as in the code you removed).

Reply:
Ok, done.